src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2.cs

```diff
@@ -628,6 +632,347 @@ public bool Verify()
             }
         }
 
+        /// <summary>
+        /// Creates a new X509 certificate from the file contents of an RFC 7468 PEM-encoded
+        /// certificate and private key.
+        /// </summary>
+        /// <param name="certPemFilePath">The path for the PEM-encoded X509 certificate.</param>
+        /// <param name="keyPemFilePath">
+        /// If specified, the path for the PEM-encoded private key.
+        /// If unspecified, the <paramref name="certPemFilePath" /> file will be used to load
+        /// the private key.
+        /// </param>
+        /// <returns>A new certificate with the private key.</returns>
+        /// <exception cref="ArgumentException">
+        /// <para>
+        ///   The contents of the file path in <paramref name="certPemFilePath" /> do not contain
+        ///   a PEM-encoded certificate, or it is malformed.
+        /// </para>
+        /// <para>-or-</para>
+        /// <para>
+        ///   The contents of the file path in <paramref name="keyPemFilePath" /> do not contain a
+        ///   PEM-encoded private key, or it is malformed.
```

Comment:
I agree that if a keyPemFilePath was specified and no key is appropriate that the right answer is to throw, but if the keyPem is not specified and cert.pem doesn't have a private key is that an intentional bad case?  I guess the fallback is to open the file with the path ctor, since it'll load the cert-PEM?

Reply:
> I guess the fallback is to open the file with the path ctor, since it'll load the cert-PEM?

I guess what you are asking is, "should these APIs on X509Cert2 support loading only the public X509 certificate"? Right now the answer is "no", since the constructor can do it, albeit in a roundabout away. For consistency's sake I went with "these always want a private key from somewhere".

Reply:
Yep, that was the question.  And the answer makes sense.